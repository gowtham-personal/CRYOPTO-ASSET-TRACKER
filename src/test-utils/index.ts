import {
  act,
  cleanup,
  fireEvent,
  render,
  renderHook,
  screen,
  waitFor,
  within,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Re-export everything
export { act, cleanup, fireEvent, render, renderHook, screen, userEvent, waitFor, within };
