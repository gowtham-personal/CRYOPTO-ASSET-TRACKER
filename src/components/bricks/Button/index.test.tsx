import { render, screen } from '@/test-utils';

import { Button, ButtonProps } from './index';

const getRenderedComponent = (args: Partial<ButtonProps> = {}) => {
  const defaultProps: ButtonProps = {
    text: 'Test Button',
    ...args,
  };
  return render(<Button {...defaultProps} />);
};

describe('Button', () => {
  it('should match snapshot with default props', () => {
    const { container } = getRenderedComponent(); // Renders with default text
    expect(container).toMatchSnapshot();
  });

  it('should render without errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    getRenderedComponent();
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render with the correct text', () => {
    getRenderedComponent({ text: 'Click me' });
    expect(screen.getByRole('button')).toHaveTextContent('Click me');
  });

  it('should render with the correct variant and color classes', () => {
    getRenderedComponent({ variant: 'outline', color: 'grey' });
    const button = screen.getByRole('button');
    expect(button).toHaveClass('border-neutral-200');
    expect(button).toHaveClass('text-neutral-600');
  });

  it('should apply custom className when provided', () => {
    getRenderedComponent({ className: 'custom-class' });
    expect(screen.getByRole('button')).toHaveClass('custom-class');
  });

  it('should render with left icon when provided', () => {
    getRenderedComponent({
      leftIcon: <span aria-hidden="true">left-icon</span>,
    });

    const button = screen.getByRole('button');
    // Check for text content that includes the icon text
    expect(button).toHaveTextContent('left-icon');
    // Verify icon appears before the button text
    expect(button.textContent).toMatch(/left-icon.*Test Button/);
  });

  it('should render with right icon when provided', () => {
    getRenderedComponent({
      rightIcon: <span aria-hidden="true">right-icon</span>,
    });

    const button = screen.getByRole('button');
    // Check for text content that includes the icon text
    expect(button).toHaveTextContent('right-icon');
    // Verify icon appears after the button text
    expect(button.textContent).toMatch(/Test Button.*right-icon/);
  });

  it('should be disabled when disabled prop is true', () => {
    getRenderedComponent({ disabled: true });
    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('should apply blue color variant correctly', () => {
    getRenderedComponent({ variant: 'solid', color: 'blue' });
    const button = screen.getByRole('button');
    expect(button).toHaveClass('bg-blue-600');
    expect(button).toHaveClass('text-white');
  });

  it('should apply red color variant correctly', () => {
    getRenderedComponent({ variant: 'solid', color: 'red' });
    const button = screen.getByRole('button');
    expect(button).toHaveClass('bg-red-600');
    expect(button).toHaveClass('text-white');
  });

  it('should apply transparent type correctly', () => {
    getRenderedComponent({ variant: 'transparent', color: 'grey' });
    const button = screen.getByRole('button');
    expect(button).toHaveClass('text-neutral-600');
    expect(button).toHaveClass('hover:bg-neutral-50');
  });
});
