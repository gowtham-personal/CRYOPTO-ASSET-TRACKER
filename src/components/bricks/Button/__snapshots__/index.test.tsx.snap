// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot with default props 1`] = `
<div>
  <button
    aria-label="Test Button button"
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-full rounded-md font-medium transition-colors shadow-[0px_1px_2px_0px_var(--color-shadow-10)] border-neutral-200 text-neutral-600 hover:bg-neutral-50 h-9 p-2 rounded-md cursor-pointer"
    data-slot="button"
    type="button"
  >
    Test Button
  </button>
</div>
`;
